UI/cmake/os-macos.cmake

```diff
@@ -15,3 +15,7 @@ target_link_libraries(obs-studio PRIVATE ${APPKIT} ${AVFOUNDATION} ${APPLICATION
 
 target_compile_options(obs-studio PRIVATE -Wno-error=float-conversion -Wno-error=implicit-int-conversion
                                           -Wno-error=shorten-64-to-32)
+
+if(CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 14.0.3)
+  target_compile_options(obs-studio PRIVATE -Wno-error=unqualified-std-cast-call)
+endif()
```

Comment:
Is this only relevant with AppleClang, or any new clang?

Reply:
The change was signed-off in clang in February last year, so it probably also affects LLVM-clang, but this file is for macOS only and you shouldn't use LLVM-clang on macOS.

Reply:
My question is basically "if someone were to try compiling with clang on another OS, would they also be running into this?"
(Not that I would want to block this PR over such a concern, for macOS it’s of course fine as-is.

Reply:
I haven't checked with llvm-clang, if the respective version contains the change, the answer would be "yes".